File: 프로그래머스/lv1/42889. 실패율/실패율.java
import java.util.*;
class Solution {
    public int[] solution(int N, int[] stages) {
        int[] answer = new int[N];
		 List<Integer> stageList = new ArrayList<>();
		 for(int stage : stages) {
			 stageList.add(stage);
		 }
		 Double[] rates = new Double[N];
		 
		 for(int i = 1; i <= N; i++) {
			 double mom = 0;
			 for(int j = 0; j < stages.length; j++) {
				 if(stages[j] >= i) mom++;
			 }
			 rates[i-1] = mom == 0? 0 :Collections.frequency(stageList, i)/mom;
		 }

		 List<Double> rateList = new ArrayList<>();
		 for(Double rate : rates) {
			 rateList.add(rate);
		 }
		 Collections.sort(rateList);
		 Collections.reverse(rateList);
		 List<Integer> ansList = new ArrayList<>();
		 for(int i = 0; i < N; i++) {
			 for(int j = 0; j < N; j++) {
				 if(rateList.get(i).equals(rates[j])) {
					if(ansList.contains(j+1)) {
						continue;
					}
					ansList.add(j+1); 
				 }
			 }
		 }
		 for(int i = 0; i < N; i++) {
			 answer[i] = ansList.get(i);
		 }
        return answer;
    }
}

File: 프로그래머스/1/42889. 실패율/실패율.js
function solution(N, stages) {
    let failureRate = [];
    stages.sort((a, b) => a - b);
    let i = 0;
    let stage = 0;
    while (++stage <= N){
        let start = i;
        while (stages[i] === stage) i++;
        const count = i - start;
        failureRate.push((count / (stages.length - start)) || 0);
    }
    return failureRate.map((v, i) => [i + 1, v]).sort((a, b) => b[1] - a[1] != 0 ? b[1] - a[1] : a[0] - b[0]).map(val => val[0]);
}